src/Migrations/Utils/Functions.php

```diff
@@ -382,4 +382,86 @@ public static function create_media_text_block(
             $html
         );
     }
+
+    /**
+     * Transform a block into an embed block.
+     *
+     * @param array $block - A parsed block.
+     * @param string $url - The embed URL.
+     * @param string $type - The embed type.
+     * @param string $provider - The embed provider.
+     *
+     * @return array - The transformed block.
+     */
+    public static function transform_block_to_embed(
+        array $block,
+        string $url,
+        string $provider,
+        string $type = 'rich',
+    ): array {
+        // phpcs:disable Generic.Files.LineLength.MaxExceeded
+        // IMPORTANT: DO NOT MODIFY THIS FORMAT!
+        $html_content = '<figure class="wp-block-embed is-type-' . $type . ' is-provider-' . $provider . ' wp-block-embed-' . $provider . ' wp-embed-aspect-16-9 wp-has-aspect-ratio"><div class="wp-block-embed__wrapper">
+        ' . $url . '
+        </div></figure>';
+        // phpcs:enable Generic.Files.LineLength.MaxExceeded
+
+        $block['blockName'] = Constants::BLOCK_EMBED;
+        $block['attrs']['type'] = $type;
+        $block['attrs']['providerNameSlug'] = $provider;
+        $block['attrs']['metadata']['name'] = $provider;
+        $block['attrs']['url'] = $url;
+        $block['innerHTML'] = $html_content;
+        $block['innerContent'][0] = $html_content;
+        $block['innerBlocks'] = [];
+
+        return $block;
+    }
+
+    /**
+     * Transform a block into a group of blocks.
+     *
+     * @param array $inner_blocks - The inner blocks for the Group block.
+     * @param string $name - The Group block name.
+     *
+     * @return array - The transformed block.
+     */
+    public static function transform_block_to_group(array $inner_blocks, string $name): array
```

Comment:
I think the logic of this function can be simplified a tiny bit:

```php
public static function transform_block_to_group(array $inner_blocks, string $name): array
    {
        $html = array (
            0 => '
        <div class="wp-block-group">',
            1 => null,
            2 => '
        ',
            3 => null,
            4 => '
        ',
            5 => null,
            6 => '
        ',
            7 => null,
            8 => '</div>
        ',
        );

        $block = [];
        $block['blockName'] = Constants::BLOCK_GROUP;
        $block['attrs']['metadata']['name'] = $name;
        $block['innerBlocks'] = $inner_blocks;
        $block['innerHTML'] = $html[0] . $html[8];
        $block['innerContent'] = $html;

        return $block;
    }
```

Reply:
I personally think that this is less clear than the previous implementation, no?

Reply:
Let's keep the current implementation then.